presto-parquet/src/main/java/io/prestosql/parquet/writer/HiveParquetPrimitiveTypeConverter.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.parquet.writer;
+
+import io.prestosql.spi.PrestoException;
+import io.prestosql.spi.type.CharType;
+import io.prestosql.spi.type.DecimalType;
+import io.prestosql.spi.type.RealType;
+import io.prestosql.spi.type.VarbinaryType;
+import io.prestosql.spi.type.VarcharType;
+import org.apache.parquet.schema.OriginalType;
+import org.apache.parquet.schema.PrimitiveType;
+import org.apache.parquet.schema.Type;
+import org.apache.parquet.schema.Types;
+
+import static io.prestosql.spi.StandardErrorCode.NOT_SUPPORTED;
+import static io.prestosql.spi.type.BigintType.BIGINT;
+import static io.prestosql.spi.type.BooleanType.BOOLEAN;
+import static io.prestosql.spi.type.DateType.DATE;
+import static io.prestosql.spi.type.DoubleType.DOUBLE;
+import static io.prestosql.spi.type.IntegerType.INTEGER;
+import static io.prestosql.spi.type.SmallintType.SMALLINT;
+import static io.prestosql.spi.type.TimestampType.TIMESTAMP;
+import static io.prestosql.spi.type.TinyintType.TINYINT;
+import static java.lang.String.format;
+import static org.apache.hadoop.hive.ql.io.parquet.serde.ParquetHiveSerDe.PRECISION_TO_BYTE_COUNT;
+import static org.apache.parquet.schema.Type.Repetition.OPTIONAL;
+
+public class HiveParquetPrimitiveTypeConverter
+        implements PrimitiveTypeConverter
+{
+    public static final HiveParquetPrimitiveTypeConverter INSTANCE = new HiveParquetPrimitiveTypeConverter();
+
+    private HiveParquetPrimitiveTypeConverter()
+    {
+    }
+
+    @Override
+    public Type getPrimitiveType(io.prestosql.spi.type.Type type, String name)
+    {
+        if (BOOLEAN.equals(type)) {
+            return Types.primitive(PrimitiveType.PrimitiveTypeName.BOOLEAN, OPTIONAL).named(name);
+        }
+        if (INTEGER.equals(type) || SMALLINT.equals(type) || TINYINT.equals(type)) {
```

Comment:
Same as Iceberg, these and `bigint` should have `as(intType(..., true))`

Reply:
Superseded.